android/src/main/java/com/balthazargronon/RCTZeroconf/ZeroconfModule.java

```diff
@@ -114,10 +114,19 @@ public void stop() {
 
     protected void sendEvent(ReactContext reactContext,
                              String eventName,
-                             @Nullable Object params) {
-        reactContext
+                             @Nullable Object params,
+                             @Nullable String errorString
+                             ) {
+        if (errorString == null){
+                    reactContext
                 .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                 .emit(eventName, params);
```

Comment:
You could use a simple ternary here to avoid repeating the same code, and send the errorString instead of the params if it's defined.

From my understanding, the errorString couldn't work because it was a String instead of an Object?

Reply:
Yes, sorry this block is wrong. I´m going to upload a fix for this ;)